zkvm/src/vm.rs

```diff
@@ -740,36 +739,25 @@ where
         Ok(())
     }
 
-    fn left_or_right<F>(&mut self, assign: F) -> Result<(), VMError>
-    where
-        F: FnOnce(&mut Contract, Predicate, Predicate) -> (),
-    {
-        let r = self.pop_item()?.to_data()?.to_predicate()?;
-        let l = self.pop_item()?.to_data()?.to_predicate()?;
+    fn select(&mut self, n: u8, k: u8) -> Result<(), VMError> {
+        if k >= n {
+            return Err(VMError::PredicateIndexInvalid)
+        }
 
+        let mut x_vec: Vec<Predicate> = Vec::with_capacity(n as usize);
+        for _ in 0..n {
+            x_vec.insert(0, self.pop_item()?.to_data()?.to_predicate()?);
+        }
         let mut contract = self.pop_item()?.to_contract()?;
         let p = &contract.predicate;
+        let pred = x_vec[k as usize].clone();
```

Comment:
per https://github.com/interstellar/slingshot/pull/227/files#r266718506 - if we pass the slice of predicates to `prove_disjunction`, then we don't need to clone this item here.